include/tvm/relax/expr.h

```diff
@@ -796,9 +798,6 @@ class MatchCast : public Binding {
 
 class VarBindingNode : public BindingNode {
  public:
-  /*! \brief The binding value. */
-  Expr value;
```

Comment:
We intentionally keep the value in the subclass, because in match clas there is extra set of struct info to be considered(that can populates new values), and it is important to keep reminding the users of this. 

The extra dispatch is worth overall for this explicitness

Reply:
Good point, and closing this PR.